src/data.rs

```diff
@@ -123,9 +125,16 @@ fn list_stacks_from_homedir() -> Result<Vec<String>, impl Error> {
 }
 
 fn sigi_path() -> String {
-    let home = env::var("HOME").or_else(|_| env::var("HOMEDRIVE")).unwrap();
-    let path = format!("{}/.local/share/sigi", home);
-    Path::new(&path).to_string_lossy().to_string()
+    let old_base = BaseDirs::new().unwrap();
+    let old_path = old_base.home_dir().join(".local").join("share").join("sigi");
+    let new_base = ProjectDirs::from("rs", "sigi-cli", "sigi").unwrap();
```

Comment:
I'm curious what you're using sigi for.

I think for the basic case of installing sigi as a CLI (and using that to manage todo lists and such) the most appropriate location is probably: https://docs.rs/directories/4.0.1/directories/struct.BaseDirs.html#method.data_local_dir

(And admittedly, using XDG conventions regardless of platform was lazy)

```suggestion
    let new_base = BaseDirs::from("sigi").unwrap();
```

If you're looking to use sigi as a library for other Rust programs that might want simple disk-persistent stacks, then "ProjectDirs" locations make more sense to me... but then I'd want to make sure the path calculation for the wrapper is public and configurable

Reply:
Also, I think it's better to go one step farther: ignore trying to use BaseDirs for how a "home" directory was defined before. Depending on how weird an environment is, it might not be the same. We can use the exact code from before as a `v1_sigi_path()` function. The same "v1_" prefix pattern is used for backwards compatibility on an earlier data format as well in the same file... Maybe not elegant, but it'd be consistent

Side note: If I keep collecting these changes, I'll probably move them to a different `data_compat.rs` library file or something.

Reply:
https://docs.rs/directories/4.0.1/directories/struct.ProjectDirs.html#method.data_dir
The `data_dir()` method of `ProjectDirs` returns the path for specific project. If you use the method from `BaseDirs` then it's required to construct the sigi specific path manually.

I agree that v1_ prefix-ing will make it more consistent.

Reply:
Oops, trying to code while sleepy... What I was going for, I think, was this:

```
    let new_base = BaseDirs::data_local_dir().join("sigi");
```

But I was misunderstanding the descriptions and semantics of BaseDirs and ProjectDirs from the project. (Also, I'm a bit unfamiliar with Mac and Windows)

Reading the docs more thoroughly, and I think you're right to choose ProjectDirs here.

https://docs.rs/directories/4.0.1/directories/struct.ProjectDirs.html#method.from

I think I should get the "qualifier" right before making this change, so I went and reserved a thing: https://sigi-cli.org

```suggestion
    let new_base = ProjectDirs::from("org.sigi-cli", "sigi-cli", "sigi").unwrap();
```

Reply:
Should be good now. Changed `v1_sigi_path()` slightly to use PathBuf to avoid converting to and from string.